client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -41,7 +42,11 @@ export class TeaListContainer extends React.Component {
 
       const revSortOrder = [...newSortOrder].reverse();
 
-      currentState[columnName] = { asc: newSortOrder, desc: revSortOrder };
+      currentState = {
+        columnName: columnName,
+        asc: newSortOrder,
+        desc: revSortOrder
+      };
     }
     this.setState({
       ...this.state,
```

Comment:
Since you're working on this function, something you might as well change in this PR: when you set state, you're setting it shallowly, meaning only the properties you provide will be changed. That means you don't have to spread `...this.state` - just leave it out entirely and all other properties will be unchanged.

Also, if you DO need to access the old state to set the new state, you shouldn't do it this way. In the Glorious React Future, state changes will be done asynchronously, meaning you won't know when state will actually be set, and `this.state` might already be out of date. Instead (not here, but in the future), you can pass a function into `setState` instead, e.g.:
```
this.setState(prevState => ({ opposite: !prevState.opposite }));
```